kettle-plugins/formats/src/main/java/org/pentaho/big/data/kettle/plugins/formats/impl/output/PvfsFileAliaser.java

```diff
@@ -0,0 +1,112 @@
+import org.pentaho.di.core.variables.VariableSpace;
+import org.pentaho.di.core.vfs.AliasedFileObject;
+import org.pentaho.di.core.vfs.KettleVFS;
+import org.pentaho.hadoop.shim.api.format.IPvfsAliasGenerator;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.FileAlreadyExistsException;
+
+/**
+ * Logic to use a temporary file for output and then copy that file to some VFS/PVFS scheme that wasn't original
+ * supoorted for the output content.
+ */
+public class PvfsFileAliaser {
+  private String finalFilePath;
+
+  private String temporaryFilePath;
+
+  private VariableSpace variableSpace;
+
+  private IPvfsAliasGenerator aliasGenerator;
+
+  private boolean isOverwriteOutput;
+
+  private LogChannelInterface log;
+
+  public PvfsFileAliaser( String finalFilePath, VariableSpace variableSpace, IPvfsAliasGenerator aliasGenerator,
+                          boolean isOverwriteOutput, LogChannelInterface log ) {
+    this.finalFilePath = finalFilePath;
+    this.variableSpace = variableSpace;
+    this.aliasGenerator = aliasGenerator;
+    this.isOverwriteOutput = isOverwriteOutput;
+    this.log = log;
+  }
+
+  public String generateAlias() throws KettleFileException, FileSystemException, FileAlreadyExistsException {
+
+    FileObject pvfsFileObject = KettleVFS.getFileObject( finalFilePath, variableSpace );
+    if ( AliasedFileObject.isAliasedFile( pvfsFileObject ) ) {
+      finalFilePath = ( (AliasedFileObject) pvfsFileObject ).getOriginalURIString();
+    }
+    //See if we need to use a another URI because the HadoopFileSystem is not supported for this URL.
+    String aliasedFile = aliasGenerator.generateAlias( finalFilePath );
+    temporaryFilePath = finalFilePath;
+    if ( aliasedFile != null ) {
+      if ( pvfsFileObject.exists() ) {
+        if ( isOverwriteOutput ) {
+          pvfsFileObject.delete();
+        } else {
+          throw new FileAlreadyExistsException( temporaryFilePath );
+        }
+      }
+      temporaryFilePath = aliasedFile;  //set the outputFile to the temporary alias file
+    }
+    return temporaryFilePath;
+  }
+
+  public void copyFileToFinalDestination() throws KettleFileException, IOException {
+    if ( aliasingIsActive() ) {
+      FileObject srcFile = KettleVFS.getFileObject( temporaryFilePath, variableSpace );
+      FileObject destFile = KettleVFS.getFileObject( finalFilePath, variableSpace );
+      try ( InputStream in = KettleVFS.getInputStream( srcFile );
+            OutputStream out = KettleVFS.getOutputStream( destFile, false ) ) {
+        IOUtils.copy( in, out );
+      }
+    }
+  }
+
+  public void deleteTempFileAndFolder() {
+    try {
+      if ( aliasingIsActive() ) {
+        FileObject srcFile = KettleVFS.getFileObject( temporaryFilePath, variableSpace );
+        srcFile.getParent().deleteAll();
+      }
+    } catch ( FileSystemException | KettleFileException e ) {
+      log.logError( e.getMessage() );
```

Comment:
Can you add the exception as the second parameter?  Otherwise we won't be able to tell where the error happened from just the message.